File: examples/python/checkpoint_consumer_oneshot.py
import os
from commit_engine import CommitEngine

CHECKPOINT_PATH = "/tmp/orders-consumer.ckpt"

def load_or_capture(engine: CommitEngine):
    if os.path.exists(CHECKPOINT_PATH):
        return engine.load_checkpoint(CHECKPOINT_PATH)
    return engine.capture_checkpoint(True)

def main():
    engine = CommitEngine(shard_count=4, inline_max=16)
    cp = load_or_capture(engine)

    items, next_cp = engine.read_from_checkpoint(cp, limit=128)

    print("items =", len(items))
    for item in items:
        print(item.seqno, item.key, item.value)

    print("next checkpoint cursors =", next_cp.cursors)

if __name__ == "__main__":
    main()

File: examples/go/checkpoint_consumer_oneshot.go
package main

import (
	"fmt"

	ce "commitengine_examples/commitengine"
)

const checkpointPath = "/tmp/orders-consumer.ckpt"

func loadOrCapture(engine *ce.Engine, path string) (*ce.Checkpoint, error) {
	cp, err := ce.LoadCheckpoint(path)
	if err == nil {
		return cp, nil
	}
	return engine.CaptureCheckpoint(true)
}

func main() {
	engine, err := ce.Open(4, 16)
	if err != nil {
		panic(err)
	}
	defer engine.Close()

	cp, err := loadOrCapture(engine, checkpointPath)
	if err != nil {
		panic(err)
	}

	items, nextCP, err := engine.ReadFromCheckpointEx(cp, 128)
	if err != nil {
		panic(err)
	}

	fmt.Printf("items=%d\n", len(items))
	for _, item := range items {
		fmt.Printf("seq=%d key=%s tombstone=%v\n", item.Seqno, string(item.Key), item.Tombstone)
	}

	if nextCP != nil {
		fmt.Printf("next checkpoint cursors=%+v\n", nextCP.Cursors)
	}
}